File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/lua/BeerArmor.lua
function GetName()
  return "Beer Armor"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TakingDamage }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == TakingDamage then
    param1 = param1 * 0.9

    if Roll(0, 99) < 5 then
      GetCurrentAttacker():ApplyAffliction("Drunk")
    end
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/BeerArmor.cs
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class BeerArmor : Affliction
    {
        public override string Name { get { return "Beer Armor"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TakingDamage, 
                };
            }
        }

        protected override void OnTakingDamage(ICharacter character, ref double damage)
        {
            damage *= .9;

            if (RollPercentage(5))
            {
                ApplyAffliction(GetCurrentAttacker(), "Drunk");
            }
        }
    }
}